git/homework.py

```diff
@@ -11,132 +7,66 @@ class OurAwesomeException(Exception):
 
 
 def is_two_object_has_same_value(first: Any, second: Any) -> bool:
-    """
-    If @first and @second has same value should return True
-    In another case should return False
-    """
-    pass
+    return first == second
 
 
 def is_two_objects_has_same_type(first: Any, second: Any) -> bool:
-    """
-    If @first and @second has same type should return True
-    In another case should return False
-    """
-    pass
+    return type(first) == type(second)
 
 
 def is_two_objects_is_the_same_objects(first: Any, second: Any) -> bool:
-    """
-    If @first and @second has same type should return True
-    In another case should return False
-    """
-    pass
+    return id(first) == id(second)
 
 
 def multiple_ints(first_value: int, second_value: int) -> int:
-    """
-    Should calculate product of all args.
-    if first_value or second_value is not int should raise ValueError
-
-    Raises:
-        ValueError
-
-    Params:
-        first_value: value for multiply
-        second_value
-    Returns:
-        Product of elements
-    """
-    pass
+    if isinstance(first_value, int) and isinstance(second_value, int):
+        return first_value * second_value
+    else:
+        raise ValueError
 
 
 def multiple_ints_with_conversion(first_value: Any, second_value: Any) -> int:
-    """
-    If possible to convert arguments to int value - convert and multiply them.
-    If it is impossible raise ValueError
-
-    Args:
-        first_value: number for multiply
-        second_value: number for multiply
-
-    Raises:
-        ValueError
-
-    Returns: multiple of two numbers.
-
-    Examples:
-        multiple_ints_with_conversion(6, 6)
-        >>> 36
-        multiple_ints_with_conversion(2, 2.0)
-        >>> 4
-        multiple_ints_with_conversion("12", 1)
-        >>> 12
-        try:
-            multiple_ints_with_conversion("Hello", 2)
-        except ValueError:
-            print("Not valid input data")
-        >>> "Not valid input data"
-    """
-    pass
+    try:
+        first_value = int(first_value)
+        second_value = int(second_value)
+        return first_value * second_value
+    except TypeError:
+        raise ValueError
 
 
 def is_word_in_text(word: str, text: str) -> bool:
-    """
-    If text contain word return True
-    In another case return False.
-
-    Args:
-        word: Searchable substring
-        text: Text for searching
-
-    Examples:
-        is_word_in_text("Hello", "Hello word")
-        >>> True
-        is_word_in_text("Glad", "Nice to meet you ")
-        >>> False
-
-    """
-    pass
+    return word in text.split()
```

Comment:
You can not use the ` split() ` method here, because the ` in ` operator can check if the substring exists in the string

Reply:
If I write `"word in text"` it will search for substring and not a word. But by function definition we should check word, not substring.
Examples:
        `is_word_in_text("Hello", "Hello word")` - True
        `is_word_in_text("Hello", "Helloword")` - False (without `.split()` it will be true also)